flows/vision-zero/cris_import/lib/mappings.py

```diff
@@ -0,0 +1,94 @@
+# This file is used to provide mappings and lists of fields for various purposes to the
+# CRIS import flow. It's designed to provide the absolute minimum of information needed.
+# Wherever data about a field can be queried from the database, the flow will do that,
+# hopefully eliminating the need to maintain this file in the future, except when we need
+# to mark a new field as editable by the VZ team.
+
+
+def get_table_map():
+    return {
+        "crash": "atd_txdot_crashes",
+        "unit": "atd_txdot_units",
+        "person": "atd_txdot_person",
+        "primaryperson": "atd_txdot_primaryperson",
+        # this table needs to be reworked to use this system
+        # "charges": "atd_txdot_charges",
+    }
+
+
+def get_key_columns():
+    return {
+        "atd_txdot_crashes": ["crash_id"],
+        "atd_txdot_units": ["crash_id", "unit_nbr"],
+        "atd_txdot_primaryperson": [
+            "crash_id",
+            "unit_nbr",
+            "prsn_nbr",
+            "prsn_type_id",
+            "prsn_occpnt_pos_id",
+        ],
+        "atd_txdot_person": [
+            "crash_id",
+            "unit_nbr",
+            "prsn_nbr",
+            "prsn_type_id",
+            "prsn_occpnt_pos_id",
+        ],
+        "atd_txdot_charges": ["crash_id", "prsn_nbr", "unit_nbr"],
+    }
+
+
+def no_override_columns():
+    return {
+        "atd_txdot_crashes": {
+            "crash_id",  # key column
+            "longitude_primary",
+            "latitude_primary",
+            "city_id",
+            "crash_speed_limit",
+            "road_type_id",
+            "traffic_cntl_id",
+            "crash_sev_id",
+            "atd_fatality_count",
+            "apd_confirmed_death_count",
+            "tot_injry_cnt",
+            "atd_txdot_crashes.sus_serious_injry_cnt",
+            "address_confirmed_primary",
+            "address_confirmed_secondary",
+            "private_dr_fl",
+            "road_constr_zone_fl",
+            "micromobility_device_flag",
+        },
+        "atd_txdot_units": {
+            "crash_id",  # key column
+            "unit_nbr",  # key column
```

Comment:
@chiaberry asked me a really good question about what is up with these `#key column` comments. I wrote:

> key columns are used here to mean “the set of columns whose combined answer are able to uniquely identify a certain record”.

> and, so if one of those were to change, you’d be inherently changing the fundamental real-life object that that record is meant to refer to. for persons for example, unit_nbr changing would mean you’re talking about an entirely different human.

> I think you have to consider the columns which are used to uniquely identify any given row as immutable in a change-comparison system, so that’s why they belong in the list of columns which can not have changes accepted without review.